grouper/plugin.py

```diff
@@ -95,3 +95,25 @@ def log_auditlog_entry(self, entry):
             entry (models.audit_log.AuditLog): just-saved log object
         """
         pass
+
+    def will_update_group_membership(self, session, group, member, **updates):
+        """
+        Called before applying changes to a group membership
+
+        Args:
+            session(sqlalchemy.orm.session.Session): database session
+            group(models.Group): affected group
+            member: affected User or Group
+            updates: the updates to the membership (active, expiration, role)
```

Comment:
nit: would it makes sense to start turning these into mypy annotations?

Reply:
totally. I was following the convention in that file, but mypy annotations make more sense.

Reply:
@herb decided to do this as a separate PR, as this one's starting to get large: #330